content/en/blog/_posts/2021-03-18-introducing-suspended-jobs.md

```diff
@@ -0,0 +1,100 @@
+---
+title: "Introducing Suspended Jobs"
+date: 2021-03-18
+slug: introducing-suspended-jobs
+---
+
+**Author:** Adhityaa Chandrasekar (Google)
+
+[Jobs](/docs/concepts/workloads/controllers/job/) are a crucial part of
+Kubernetes API. While other kinds of workloads such as [Deployments](/docs/concepts/workloads/controllers/deployment/),
+[ReplicaSets](/docs/concepts/workloads/controllers/replicaset/),
+[StatefulSets](/docs/concepts/workloads/controllers/statefulset/), and
+[DaemonSets](/docs/concepts/workloads/controllers/daemonset/)
+solve use-cases that require Pods to run forever, Jobs are useful when Pods need
+to run to completion. Commonly used in parallel batch processing, Jobs can be
+used in a variety of applications ranging from video rendering and database
+maintenance to sending bulk emails and scientific computing.
+
+While the amount of parallelism and the conditions for Job completion are
+configurable, the Kubernetes API lacked the ability to suspend and resume Jobs.
+This is often desired when cluster resources are limited and a higher priority
+Job needs to execute in the place of another Job. Deleting the lower priority
+Job is a poor workaround as Pod completion history and other metrics associated
+with the Job will be lost.
+
+With the recent Kubernetes 1.21 release, you will be able to suspend a Job by
+updating its Job spec. The feature is currently in **alpha** and requires you to
+enable the `SuspendJob` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+on the [API server](/docs/reference/command-line-tools-reference/kube-apiserver/)
+and the [controller manager](/docs/reference/command-line-tools-reference/kube-controller-manager/)
+in order to use it.
+
+## API changes
+
+A new boolean field `suspend` is introduced in the Job spec API. Let's say I
+create the following Job.
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: my-job
+spec:
+  suspend: true
+  parallelism: 2
+  completions: 10
+  template:
+    spec:
+      containers:
+      - name: my-container
+        image: busybox
+        command: ["sleep", "5"]
+      restartPolicy: Never
+```
+
+Jobs are not suspended by default, so I need to explicitly set the `suspend`
+field to true in the above Job spec. The Job controller will now refrain from
+creating Pods until I'm ready to start the Job, which I can do by updating the
+field to false.
+
+If I update this field from false back to true again, the Job controller will
+terminate all active Pods it created. This is done by sending the Pods a SIGTERM
+signal; the [graceful termination period](/docs/concepts/workloads/pods/pod-lifecycle/#pod-termination)
+defined in the Pod spec will be honoured and Pods terminated this way will not
+be counted as failures by the Job controller.
```

Comment:
I would explicitly indicate that success and failed pods will continue to exist, and that upon unsuspension they will count against job completion. This will set the stage for the next section discussing the where this is useful.

Reply:
fair point, I've added a paragraph. PTAL, thanks.